core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -93,8 +95,9 @@ protected void waitUntilReady() {
 
     private boolean shouldCheckWithCommand() {
         // Special case for Docker for Mac, see #160
-        if(!DockerClientFactory.instance().isUsing(DockerMachineClientProviderStrategy.class)
-                && System.getProperty("os.name").toLowerCase().contains("mac")) {
+        if ((DockerClientFactory.instance().isUsing(ProxiedUnixSocketClientProviderStrategy.class) ||
```

Comment:
I think previous code (*not* docker machine *and* Mac) should cover both cases. Also, this change has a bug because || has lower priority than &&